vm/systemSmartContracts/staking.go

```diff
@@ -499,15 +508,26 @@ func (r *stakingSC) processStake(blsKey []byte, registrationData *StakedDataV2_0
 		return nil
 	}
 
-	r.addToStakedNodes()
-	registrationData.Staked = true
-	registrationData.StakedNonce = r.eei.BlockChainHook().CurrentNonce()
-	registrationData.UnStakedEpoch = core.DefaultUnstakedEpoch
-	registrationData.UnStakedNonce = 0
+	err := r.removeFromWaitingList(blsKey)
+	if err != nil {
+		r.eei.AddReturnMessage("error while removing from waiting")
+		return err
+	}
+	r.addToStakedNodes(1)
+	r.activeStakingFor(registrationData)
 
 	return nil
 }
 
+func (r *stakingSC) activeStakingFor(stakingData *StakedDataV2_0) {
+	stakingData.RegisterNonce = r.eei.BlockChainHook().CurrentNonce()
```

Comment:
If you set it here now you can move line 498 after line 505 ?

Reply:
No. Registration NONCE is needed when getting into the waiting list as well

Reply:
No you can't. As you need the registration NONCE even when it got into the waiting list.